consumer.go

```diff
@@ -642,7 +642,7 @@ func (bc *brokerConsumer) handleResponses() {
 			Logger.Printf("consumer/%s/%d shutting down because %s\n", child.topic, child.partition, result)
 			close(child.trigger)
 			delete(bc.subscriptions, child)
-		case ErrUnknownTopicOrPartition, ErrNotLeaderForPartition, ErrLeaderNotAvailable:
+		case ErrUnknownTopicOrPartition, ErrNotLeaderForPartition, ErrLeaderNotAvailable, ErrReplicaNotAvailable:
```

Comment:
Ya, that's true I think? It's not super-clear from the protocol spec what to do with `ReplicaNotAvailable`. It says of the error "this can be safely ignored" but also marks it as not-retriable, so... treat it like success? Hard fail? @edenhill could you clarify please?